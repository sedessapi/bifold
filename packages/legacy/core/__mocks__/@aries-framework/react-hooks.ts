/* eslint-disable @typescript-eslint/no-explicit-any */
import { LegacyIndyCredentialFormat } from '@aries-framework/anoncreds'
import {
  BasicMessageRecord,
  CredentialExchangeRecord,
  CredentialProtocolOptions,
  ProofExchangeRecord,
} from '@aries-framework/core'

const useCredentials = jest.fn().mockReturnValue({ records: [] } as any)
const useProofs = jest.fn().mockReturnValue({ records: [] } as any)
const useCredentialByState = jest.fn().mockReturnValue([] as CredentialExchangeRecord[])
const useProofByState = jest.fn().mockReturnValue([] as ProofExchangeRecord[])
const useBasicMessagesByConnectionId = jest.fn().mockReturnValue([] as BasicMessageRecord[])
const useBasicMessages = jest.fn().mockReturnValue({ records: [] as BasicMessageRecord[] })
const mockCredentialModule = {
  acceptOffer: jest.fn(),
  declineOffer: jest.fn(),
  getFormatData: jest
    .fn()
    .mockReturnValue(
      Promise.resolve({} as CredentialProtocolOptions.GetCredentialFormatDataReturn<[LegacyIndyCredentialFormat]>)
    ),
}
const mockProofModule = {
  getCredentialsForRequest: jest.fn(),
  acceptRequest: jest.fn(),
  declineRequest: jest.fn(),
  getFormatData: jest.fn(),
  findRequestMessage: jest.fn(),
  requestProof: jest.fn(),
  update: jest.fn(),
}

const mockMediationRecipient = {
  initiateMessagePickup: jest.fn(),
}
const mockOobModule = {
  findById: jest.fn().mockReturnValue(Promise.resolve(null)),
  createInvitation: jest.fn(),
  toUrl: jest.fn(),
}
const mockBasicMessageRepository = {
  update: jest.fn(),
}
const mockAgentContext = {
  dependencyManager: {
    resolve: jest.fn().mockReturnValue(mockBasicMessageRepository),
  },
}
const useAgent = () => ({
  agent: {
    credentials: mockCredentialModule,
    proofs: mockProofModule,
    mediationRecipient: mockMediationRecipient,
    oob: mockOobModule,
    context: mockAgentContext,
  },
})
const useCredentialById = jest.fn()
const useProofById = jest.fn()
const useConnectionById = jest.fn()
const useConnections = jest.fn()

export {
  useAgent,
  useConnectionById,
  useCredentials,
  useProofs,
  useCredentialById,
  useCredentialByState,
  useProofById,
  useProofByState,
  useConnections,
  useBasicMessages,
  useBasicMessagesByConnectionId,
}
