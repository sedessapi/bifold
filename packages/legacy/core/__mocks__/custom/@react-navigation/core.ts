const navigate = jest.fn()
const dispatch = jest.fn()
const navigation = {
  __timestamp: process.hrtime(),
  navigate,
  setOptions: jest.fn(),
  getParent: () => {return {
    navigate,
    dispatch,
  }},
  getState: jest.fn(() => ({
    index: jest.fn(),
  })),
  goBack: jest.fn(),
  pop: jest.fn(),
  reset: jest.fn(),
  isFocused: () => true,
  dispatch,
}

const useNavigation = () => {
  return navigation
}

const useIsFocused = () => {
  return true
}

export { useNavigation, useIsFocused }
