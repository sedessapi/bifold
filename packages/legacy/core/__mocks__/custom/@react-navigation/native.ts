// import { useFocusEffect } from '@react-navigation/native'

const useFocusEffect = jest.fn()
const createNavigatorFactory = jest.fn()

const CommonActions = {
  reset: jest.fn(),
}

const useIsFocused = () => {
  return true
}

export { useIsFocused, useFocusEffect, createNavigatorFactory, CommonActions }
