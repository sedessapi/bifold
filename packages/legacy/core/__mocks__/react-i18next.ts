const mockT = jest.fn((key: string) => key)
const useTranslation = () => {
  return {
    t: mockT,
    i18n: {
      language: 'en',
    },
  }
}

export { useTranslation }
