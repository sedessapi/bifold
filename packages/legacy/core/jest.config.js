module.exports = {
  preset: 'react-native',
  testTimeout: 10000,
  setupFiles: ['<rootDir>/jestSetup.js'],
  setupFilesAfterEnv: ['@testing-library/jest-native/extend-expect'],
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'json', 'node'],
  moduleNameMapper: {
    '\\.(jpg|ico|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$':
      '<rootDir>/__mocks__/file.js',
    '\\.(css|less)$': '<rootDir>/__mocks__/style.js',
    axios: require.resolve('axios'),
    'react-i18next': '<rootDir>/__mocks__/react-i18next.ts',
    '^uuid$': require.resolve('uuid'),
    '@aries-framework/core': require.resolve('@aries-framework/core'),
    '@aries-framework/anoncreds': require.resolve('@aries-framework/anoncreds'),
  },
  transform: {
    '^.+\\.(js|jsx|ts|tsx)$': 'babel-jest',
  },
  transformIgnorePatterns: [
    'node_modules\\/(?!(.*react-native.*)|(uuid)|(@aries-framework\\/core)|(@aries-framework\\/anoncreds))',
  ],
  testRegex: '(/__tests__/.*|(\\.|/)(test|spec))\\.[jt]sx?$',
  testPathIgnorePatterns: [
    '\\.snap$',
    '<rootDir>/node_modules/',
    '<rootDir>/lib',
    '<rootDir>/__tests__/contexts/',
    '<rootDir>/__tests__/helpers/',
  ],
  coveragePathIgnorePatterns: [
    '/node_modules/',
    '/__tests__/',
    '<rootDir>/App/navigators/defaultStackOptions.tsx',
    '<rootDir>/App/defaultConfiguration.ts',
    '<rootDir>/App/components/buttons/InfoIcon.tsx',
    '<rootDir>/App/hooks/deep-links.ts',
  ],
  cacheDirectory: '.jest/cache',
  snapshotFormat: {
    escapeString: true,
    printBasicPrototype: true,
  },
}
