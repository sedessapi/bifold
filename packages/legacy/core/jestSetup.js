/* eslint-disable no-undef */
// eslint-disable-next-line import/no-extraneous-dependencies
import 'reflect-metadata'
import 'react-native-gesture-handler/jestSetup'
import mockRNCNetInfo from '@react-native-community/netinfo/jest/netinfo-mock.js'
import mockRNLocalize from 'react-native-localize/mock'

jest.mock('@react-native-community/netinfo', () => mockRNCNetInfo)
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper')
jest.mock('react-native/Libraries/EventEmitter/NativeEventEmitter')
jest.mock('react-native-localize', () => mockRNLocalize)
