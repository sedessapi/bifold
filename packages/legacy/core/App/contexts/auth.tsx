// NOTE: We need to import these to be able to use the AskarWallet in this file.
import '@hyperledger/aries-askar-react-native'
// eslint-disable-next-line import/no-extraneous-dependencies
import 'reflect-metadata'

import { AskarWallet } from '@aries-framework/askar'
import { ConsoleLogger, LogLevel, SigningProviderRegistry } from '@aries-framework/core'
import { useAgent } from '@aries-framework/react-hooks'
import { agentDependencies } from '@aries-framework/react-native'
import React, { createContext, useContext, useState } from 'react'
import { useTranslation } from 'react-i18next'
import { DeviceEventEmitter } from 'react-native'

import { EventTypes } from '../constants'
import { DispatchAction } from '../contexts/reducers/store'
import { useStore } from '../contexts/store'
import {
  secretForPIN,
  storeWalletSecret,
  loadWalletSecret,
  loadWalletSalt,
  isBiometricsActive,
  wipeWalletKey,
} from '../services/keychain'
import { WalletSecret } from '../types/security'
import { hashPIN } from '../utils/crypto'
import { migrateToAskar, didMigrateToAskar } from '../utils/migration'

export interface AuthContext {
  checkPIN: (PIN: string) => Promise<boolean>
  getWalletCredentials: () => Promise<WalletSecret | undefined>
  removeSavedWalletSecret: () => void
  disableBiometrics: () => Promise<void>
  setPIN: (PIN: string) => Promise<void>
  commitPIN: (useBiometry: boolean) => Promise<boolean>
  isBiometricsActive: () => Promise<boolean>
  rekeyWallet: (oldPin: string, newPin: string, useBiometry?: boolean) => Promise<boolean>
}

export const AuthContext = createContext<AuthContext>(null as unknown as AuthContext)

export const AuthProvider: React.FC<React.PropsWithChildren> = ({ children }) => {
  const [walletSecret, setWalletSecret] = useState<WalletSecret>()
  const [store, dispatch] = useStore()
  const { agent } = useAgent()
  const { t } = useTranslation()

  const setPIN = async (PIN: string): Promise<void> => {
    const secret = await secretForPIN(PIN)
    await storeWalletSecret(secret)
  }

  const getWalletCredentials = async (): Promise<WalletSecret | undefined> => {
    if (walletSecret && walletSecret.key) {
      return walletSecret
    }

    const { secret, err } = await loadWalletSecret(
      t('Biometry.UnlockPromptTitle'),
      t('Biometry.UnlockPromptDescription')
    )

    DeviceEventEmitter.emit(EventTypes.BIOMETRY_ERROR, err !== undefined)

    if (!secret) {
      return
    }
    setWalletSecret(secret)

    return secret
  }

  const commitPIN = async (useBiometry: boolean): Promise<boolean> => {
    const secret = await getWalletCredentials()
    if (!secret) {
      return false
    }
    // set did authenticate to true if we can get wallet credentials
    dispatch({
      type: DispatchAction.DID_AUTHENTICATE,
    })
    if (useBiometry) {
      await storeWalletSecret(secret, useBiometry)
    } else {
      // erase wallet key if biometrics is disabled
      await wipeWalletKey(useBiometry)
    }
    return true
  }

  const checkPIN = async (PIN: string): Promise<boolean> => {
    try {
      const secret = await loadWalletSalt()

      if (!secret || !secret.salt) {
        return false
      }

      const hash = await hashPIN(PIN, secret.salt)

      if (!didMigrateToAskar(store.migration)) {
        await migrateToAskar(secret.id, hash)
        dispatch({
          type: DispatchAction.DID_MIGRATE_TO_ASKAR,
        })
      }

      // NOTE: a custom wallet is used to check if the wallet key is correct. This is different from the wallet used in the rest of the app.
      // We create an AskarWallet instance and open the wallet with the given secret.
      const askarWallet = new AskarWallet(
        new ConsoleLogger(LogLevel.off),
        new agentDependencies.FileSystem(),
        new SigningProviderRegistry([])
      )
      await askarWallet.open({
        id: secret.id,
        key: hash,
      })

      await askarWallet.close()

      const fullSecret = await secretForPIN(PIN, secret.salt)
      setWalletSecret(fullSecret)
      return true
    } catch (e) {
      return false
    }
  }

  const removeSavedWalletSecret = () => {
    setWalletSecret(undefined)
  }

  const disableBiometrics = async () => {
    await wipeWalletKey(true)
  }

  const rekeyWallet = async (oldPin: string, newPin: string, useBiometry?: boolean): Promise<boolean> => {
    try {
      // argon2.hash can sometimes generate an error
      const secret = await loadWalletSalt()
      if (!secret) {
        return false
      }
      const oldHash = await hashPIN(oldPin, secret.salt)
      const newSecret = await secretForPIN(newPin)
      const newHash = await hashPIN(newPin, newSecret.salt)
      if (!newSecret.key) {
        return false
      }

      await agent?.wallet.close()
      await agent?.wallet.rotateKey({ id: secret.id, key: oldHash, rekey: newHash })
      await storeWalletSecret(newSecret, useBiometry)
      setWalletSecret(newSecret)
    } catch {
      return false
    }
    return true
  }

  return (
    <AuthContext.Provider
      value={{
        checkPIN,
        getWalletCredentials,
        removeSavedWalletSecret,
        disableBiometrics,
        commitPIN,
        setPIN,
        isBiometricsActive,
        rekeyWallet,
      }}
    >
      {children}
    </AuthContext.Provider>
  )
}

export const useAuth = () => useContext(AuthContext)
