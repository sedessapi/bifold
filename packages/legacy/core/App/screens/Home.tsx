import { useIsFocused } from '@react-navigation/native'
import { StackScreenProps } from '@react-navigation/stack'
import React, { useEffect, useState } from 'react'
import { useTranslation } from 'react-i18next'
import { FlatList, View, StyleSheet } from 'react-native'

import NotificationListItem, { NotificationType } from '../components/listItems/NotificationListItem'
import NoNewUpdates from '../components/misc/NoNewUpdates'
import AppGuideModal from '../components/modals/AppGuideModal'
import { useConfiguration } from '../contexts/configuration'
import { DispatchAction } from '../contexts/reducers/store'
import { useStore } from '../contexts/store'
import { useTheme } from '../contexts/theme'
import { useTour } from '../contexts/tour/tour-context'
import { HomeStackParams, Screens } from '../types/navigators'
import { TourID } from '../types/tour'

type HomeProps = StackScreenProps<HomeStackParams, Screens.Home>

const Home: React.FC<HomeProps> = () => {
  const {
    useCustomNotifications,
    enableTours: enableToursConfig,
    homeFooterView: HomeFooterView,
    homeHeaderView: HomeHeaderView,
  } = useConfiguration()
  const { notifications } = useCustomNotifications()
  const { t } = useTranslation()

  const { ColorPallet } = useTheme()
  const [store, dispatch] = useStore()
  const { start } = useTour()
  const [showTourPopup, setShowTourPopup] = useState(false)
  const screenIsFocused = useIsFocused()

  const styles = StyleSheet.create({
    flatlist: {
      marginBottom: 35,
    },
    noNewUpdatesContainer: {
      paddingHorizontal: 20,
      paddingVertical: 20,
      backgroundColor: ColorPallet.brand.secondaryBackground,
    },
  })

  const DisplayListItemType = (item: any): React.ReactNode => {
    let component: React.ReactNode
    if (item.type === 'BasicMessageRecord') {
      component = <NotificationListItem notificationType={NotificationType.BasicMessage} notification={item} />
    } else if (item.type === 'CredentialRecord') {
      let notificationType = NotificationType.CredentialOffer
      if (item.revocationNotification) {
        notificationType = NotificationType.Revocation
      }
      component = <NotificationListItem notificationType={notificationType} notification={item} />
    } else if (item.type === 'CustomNotification') {
      component = <NotificationListItem notificationType={NotificationType.Custom} notification={item} />
    } else {
      component = <NotificationListItem notificationType={NotificationType.ProofRequest} notification={item} />
    }
    return component
  }

  useEffect(() => {
    const shouldShowTour = enableToursConfig && store.tours.enableTours && !store.tours.seenHomeTour

    if (shouldShowTour && screenIsFocused) {
      if (store.tours.seenToursPrompt) {
        dispatch({
          type: DispatchAction.UPDATE_SEEN_HOME_TOUR,
          payload: [true],
        })
        start(TourID.HomeTour)
      } else {
        dispatch({
          type: DispatchAction.UPDATE_SEEN_TOUR_PROMPT,
          payload: [true],
        })
        setShowTourPopup(true)
      }
    }
  }, [screenIsFocused])

  const onCTAPressed = () => {
    setShowTourPopup(false)
    dispatch({
      type: DispatchAction.ENABLE_TOURS,
      payload: [true],
    })
    dispatch({
      type: DispatchAction.UPDATE_SEEN_HOME_TOUR,
      payload: [true],
    })
    start(TourID.HomeTour)
  }

  const onDismissPressed = () => {
    setShowTourPopup(false)
    dispatch({
      type: DispatchAction.ENABLE_TOURS,
      payload: [false],
    })
  }

  return (
    <>
      <FlatList
        style={styles.flatlist}
        showsVerticalScrollIndicator={false}
        scrollEnabled={notifications?.length > 0 ? true : false}
        decelerationRate="fast"
        ListEmptyComponent={() => (
          <View style={styles.noNewUpdatesContainer}>
            <NoNewUpdates />
          </View>
        )}
        ListHeaderComponent={() => <HomeHeaderView />}
        ListFooterComponent={() => <HomeFooterView />}
        data={notifications}
        keyExtractor={(item) => item.id}
        renderItem={({ item, index }) => (
          <View
            style={{
              paddingHorizontal: 20,
              paddingTop: index === 0 ? 20 : 0,
              paddingBottom: index === notifications.length - 1 ? 20 : 10,
              backgroundColor: ColorPallet.brand.secondaryBackground,
            }}
          >
            {DisplayListItemType(item)}
          </View>
        )}
      />
      {showTourPopup && (
        <AppGuideModal
          title={t('Tour.GuideTitle')}
          description={t('Tour.WouldYouLike')}
          onCallToActionPressed={onCTAPressed}
          onCallToActionLabel={t('Tour.UseAppGuides')}
          onSecondCallToActionPressed={onDismissPressed}
          onSecondCallToActionLabel={t('Tour.DoNotUseAppGuides')}
          onDismissPressed={onDismissPressed}
        />
      )}
    </>
  )
}

export default Home
