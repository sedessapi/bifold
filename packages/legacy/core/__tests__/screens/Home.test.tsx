
jest.mock('@react-navigation/core', () => {
  return require('../../__mocks__/custom/@react-navigation/core')
})
jest.mock('@react-navigation/native', () => {
  return require('../../__mocks__/custom/@react-navigation/native')
})

jest.mock('@aries-framework/react-hooks', ()=>({
  ...jest.requireActual('@aries-framework/react-hooks'),
  useBasicMessages: jest.fn().mockReturnValue({records: []}),
  useCredentialByState: jest.fn().mockReturnValue([]),
  useProofByState: jest.fn().mockReturnValue([]),
  useAgent: jest.fn(),
  useConnectionById: jest.fn(),
}))
import {
  BasicMessageRecord,
  BasicMessageRole,
  CredentialExchangeRecord as CredentialRecord,
  CredentialState,
  ProofExchangeRecord,
  ProofState,
} from '@aries-framework/core'
//import { useBasicMessages, useCredentialByState, useProofByState, useAgent } from '@aries-framework/react-hooks'
import { useNavigation } from '@react-navigation/core'
import { act, fireEvent, render } from '@testing-library/react-native'

import React from 'react'
import { useBasicMessages, useCredentialByState, useProofByState, useAgent, useConnectionById } from '@aries-framework/react-hooks'

// eslint-disable-next-line import/no-named-as-default
import { ConfigurationContext } from '../../App/contexts/configuration'
import Home from '../../App/screens/Home'
import { testIdWithKey } from '../../App/utils/testable'
import configurationContext from '../contexts/configuration'

import { useTranslation } from 'react-i18next'

describe('displays a home screen', () => {
  beforeEach(() => {
  })

  test('renders correctly', () => {
    const tree = render(
      <ConfigurationContext.Provider value={configurationContext}>
        <Home route={{} as any} navigation={useNavigation()} />
      </ConfigurationContext.Provider>
    )

    expect(tree).toMatchSnapshot()
  })

  /**
   * Scenario: Home Screen without any pending notification
   * Given wallet has successfully loaded
   * When the holder selects the "Home" button in the main navigation bar
   * Then the Home Screen is displayed
   * TODO:(jl) Good enough to be captured by the snapshot?
   */
  test('defaults to no notifications', async () => {
    const { findByText } = render(
      <ConfigurationContext.Provider value={configurationContext}>
        <Home route={{} as any} navigation={useNavigation()} />
      </ConfigurationContext.Provider>
    )
    const notificationLabel = await findByText('Home.NoNewUpdates')

    expect(notificationLabel).toBeTruthy()
  })
})

describe('with a notifications module, when an issuer sends a credential offer', () => {
  const testCredentialRecords: CredentialRecord[] = [
    new CredentialRecord({
      threadId: '1',
      state: CredentialState.OfferReceived,
      protocolVersion: 'v1',
    }),
  ]
  const testProofRecords: ProofExchangeRecord[] = [
    new ProofExchangeRecord({
      threadId: '2',
      state: ProofState.RequestReceived,
      protocolVersion: 'v1',
    }),
    new ProofExchangeRecord({
      threadId: '3',
      state: ProofState.Done,
      protocolVersion: 'v1',
    }),
  ]
  const testBasicMessages: BasicMessageRecord[] = [
    new BasicMessageRecord({
      threadId: '1',
      connectionId: '1',
      role: BasicMessageRole.Receiver,
      content: 'Hello',
      sentTime: '20200303',
    }),
    new BasicMessageRecord({
      threadId: '2',
      connectionId: '1',
      role: BasicMessageRole.Receiver,
      content: 'Hi',
      sentTime: '20200303',
    }),
  ]

  beforeEach(() => {
    jest.resetAllMocks()
    // @ts-ignore
    useBasicMessages.mockReturnValue({records: testBasicMessages})

    // @ts-ignore
    useProofByState.mockReturnValue(testProofRecords)
    // @ts-ignore
    useCredentialByState.mockReturnValue(testCredentialRecords)

    // @ts-ignore
    useAgent.mockReturnValue({})

    // @ts-ignore
    useConnectionById.mockReturnValue({theirLabel:'ACME'})
  })

  /**
   * Scenario: Holder receives a credential offer
   * Given the holder has a connection with an issuer
   * When the issuer sends a credential offer
   * Then the credential offer will arrive in the form of a notification in the home screen
   */
  test('notifications are displayed', async () => {
    const { findAllByTestId } = render(
      <ConfigurationContext.Provider value={configurationContext}>
        <Home route={{} as any} navigation={useNavigation()} />
      </ConfigurationContext.Provider>
    )

    const flatListInstance = await findAllByTestId(testIdWithKey('NotificationListItem'))

    expect(flatListInstance).toHaveLength(4)
  })

  /**
   * Scenario: Holder selects a credential offer
   * Given the holder has received a credential offer from a connected issuer
   * When the holder selects the credential offer
   * When the holder is taken to the credential offer screen/flow
   */
  test('touch notification triggers navigation correctly I', async () => {
    const navigation = useNavigation()
    const view = render(
      <ConfigurationContext.Provider value={configurationContext}>
        <Home route={{} as any} navigation={navigation as any} />
      </ConfigurationContext.Provider>
    )

    const button = await view.findByTestId(testIdWithKey('ViewOffer'))
    

    expect(button).toBeDefined()
    //view.debug()
    await act(() => {
      fireEvent(button, 'press')
    })
    
    expect(navigation.navigate).toHaveBeenCalledTimes(1)
    expect(navigation.navigate).toHaveBeenCalledWith('Notifications Stack', {
      screen: 'Credential Offer',
      params: { credentialId: testCredentialRecords[0].id },
    })
  })

  /**
   * Scenario: Holder selects a proof request
   * Given the holder has received a proof request from a connected verifier
   * When the holder selects the proof request
   * When the holder is taken to the proof request screen/flow
   */
  test('touch notification triggers navigation correctly II', async () => {
    const { findByTestId } = render(
      <ConfigurationContext.Provider value={configurationContext}>
        <Home route={{} as any} navigation={useNavigation()} />
      </ConfigurationContext.Provider>
    )

    const button = await findByTestId(testIdWithKey('ViewProofRecord'))
    const navigation = useNavigation()

    expect(button).toBeDefined()

    await act(() => {
      fireEvent(button, 'press')
    })

    expect(navigation.navigate).toHaveBeenCalledTimes(1)

    expect(navigation.navigate).toHaveBeenCalledWith('Notifications Stack', {
      screen: 'Proof Request',
      params: { proofId: testProofRecords[0].id },
    })
  })

  /**
   * Scenario: Holder selects a proof request
   * Given the holder has received a proof request from a connected verifier
   * When the holder selects the proof request
   * When the holder is taken to the proof request screen/flow
   */
  test('touch notification triggers navigation correctly III', async () => {
    const { findByTestId } = render(
      <ConfigurationContext.Provider value={configurationContext}>
        <Home route={{} as any} navigation={useNavigation()} />
      </ConfigurationContext.Provider>
    )

    const button = await findByTestId(testIdWithKey('ViewProofRecordReceived'))
    const navigation = useNavigation()

    expect(button).toBeDefined()

    act(() => {
      fireEvent(button, 'press')
    })

    expect(navigation.navigate).toHaveBeenCalledTimes(1)
    expect(navigation.navigate).toHaveBeenCalledWith('Contacts Stack', {
      screen: 'Proof Details',
      params: { recordId: testProofRecords[1].id, isHistory: true },
    })
  })

  /**
   * Scenario: Holder selects a basic message notification
   * Given the holder has received a message from a contact
   * When the holder taps the View message button
   * The holder is taken to the chat screen for that contact
   */
  test('touch notification triggers navigation correctly IV', async () => {
    const { findByTestId } = render(
      <ConfigurationContext.Provider value={configurationContext}>
        <Home route={{} as any} navigation={useNavigation()} />
      </ConfigurationContext.Provider>
    )
    await act(async () => {
      const button = await findByTestId(testIdWithKey('ViewBasicMessage'))
      expect(button).toBeDefined()
      fireEvent(button, 'press')
    })
    const navigation = useNavigation()
    expect(navigation.getParent()?.navigate).toHaveBeenCalledTimes(1)
    expect(navigation.getParent()?.navigate).toHaveBeenCalledWith('Contacts Stack', {
      screen: 'Chat',
      params: { connectionId: '1' },
    })
  })
})
