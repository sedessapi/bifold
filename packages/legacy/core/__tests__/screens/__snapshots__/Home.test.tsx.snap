// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays a home screen renders correctly 1`] = `
<RCTScrollView
  ListEmptyComponent={[Function]}
  ListFooterComponent={[Function]}
  ListHeaderComponent={[Function]}
  data={Array []}
  decelerationRate="fast"
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEnabled={false}
  scrollEventThrottle={50}
  showsVerticalScrollIndicator={false}
  stickyHeaderIndices={Array []}
  style={
    Object {
      "marginBottom": 35,
    }
  }
  viewabilityConfigCallbackPairs={Array []}
>
  <View>
    <View
      collapsable={false}
      onLayout={[Function]}
    />
    <View
      style={
        Object {
          "backgroundColor": "#313132",
          "paddingHorizontal": 20,
          "paddingVertical": 20,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#313132",
            "borderColor": "#0099FF",
            "borderRadius": 5,
            "borderWidth": 1,
            "padding": 10,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "alignSelf": "center",
                "marginRight": 10,
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#0099FF",
                    "fontSize": 30,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              î¢Ž
            </Text>
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 18,
                  "fontWeight": "normal",
                },
                Object {
                  "flexShrink": 1,
                },
              ]
            }
            testID="com.ariesbifold:id/NoNewUpdates"
          >
            Home.NoNewUpdates
          </Text>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
    />
  </View>
</RCTScrollView>
`;
